### PR TITLE
join room basic testing added

### PR DESCRIPTION
**Pull request by beris147**
`npm test` #22 
Screenshot 2021-04-21 13:10:57

Basically JoinRoom component should not allow us to make a join-room petition if we do not provide a username and a room id. 
If we provide a valid room id and a valid username we should be able to join the room. 

Error handling testing should be provided when added to the component.

### CODE REVIEW

File: client/src/components/__tests__/JoinRoom.test.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    const username = 'owner';
+    const roomid = 'my-id';
+    let fakeRoom: RoomT = {
+      id: roomid,
+      users: [ ],
+      owner: '',
+      round: 1,
+      time: 30,
+    };
+    socket.on('join-room', (data: JoinRoomRequestT) => {
```
Comment (GustavoMeza) [inline client/src/components/__tests__/JoinRoom.test.js:111]: nice, mocking the server

File: client/src/components/__tests__/JoinRoom.test.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      fakeRoom.users.push(data.username);
+      serverSocket.emit('joined-room', fakeRoom);
+    });
+    socket.on('joined-room', (room: RoomT) => {
+      expect(room).toBe(fakeRoom);
+      expect(room.users).toContain(username);
```
Comment (GustavoMeza) [inline client/src/components/__tests__/JoinRoom.test.js:117]: Looks like you are testing if your mock server implementation updates the information as expected.
There's nothing wrong with that, but maybe you want to test instead for a `join-room` message to be sent.
One way to do it is to mock the `socket.on` to push the events into an array. Then expect your event to be there. 
There's probably better ways though.

File: client/src/components/__tests__/JoinRoom.test.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    const userNameInput = screen.getByTestId('username');
+    const roomIdInput = screen.getByTestId('roomid');
+    expect(joinButton).toBeDisabled();
+    fireEvent.change(userNameInput, { target: { value: 'username' } });
+    expect(joinButton).toBeDisabled();
+    fireEvent.change(roomIdInput, { target: { value: 'roomid' } });
+    expect(joinButton).toBeEnabled();
+  });
+  
+  test('join room sockets test', (): void => {
```
Comment (GustavoMeza) [inline client/src/components/__tests__/JoinRoom.test.js:101]: It's generally recommended to have your tests names to describe some behavior.
What is the behavior you are looking in this test?

**Review by GustavoMeza — COMMENTED** [review]
Overall looks good. I do have some suggestions.

File: client/src/components/__tests__/JoinRoom.test.js (hunk 2 above)
Comment (beris147) [reply client/src/components/__tests__/JoinRoom.test.js:117]: I am trying to test if the front end emitted the `join-room` petition if we received a `joined-room` then it worked. I am mocking the socketServer functionality. I tried to not override that behavior, but I dunno if this is a good way to test it.

**Pull request merged**